.github/workflows/release-chart.yml

```diff
@@ -0,0 +1,51 @@
+name: Release metrics adapter chart
+on:
+  workflow_dispatch:
+    inputs:
+      prerelease:
+        description: 'This is a pre-release'
+        required: false
+        default: false
+        type: boolean
+
+jobs:
+  # Sometimes chart-releaser might fetch an outdated index.yaml from gh-pages, causing a WAW hazard on the repo
+  # This job checks the remote file is up to date with the local one on release
+  validate-gh-pages-index:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: gh-pages
+      - name: Download remote index file and check equality
+        run: |
+          curl -vsSL https://newrelic.github.io/newrelic-k8s-metrics-adapter/index.yaml > index.yaml.remote
+          LOCAL="$(md5sum < index.yaml)"
+          REMOTE="$(md5sum < index.yaml.remote)"
+          echo "$LOCAL" = "$REMOTE"
+          test "$LOCAL" = "$REMOTE"
+
+  release:
+    runs-on: ubuntu-latest
+    needs: [ validate-gh-pages-index ]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Change chart's version if this run is a pre-release
+        if: ${{ github.event.input.prerelease }}
+        run: |
+          sed -i 's|^version: \(.*\)|version: \1-pre|' ./charts/newrelic-k8s-metrics-adapter/Chart.yaml
+          sed -i 's|^appVersion: \(.*\)|appVersion: \1-pre|' ./charts/newrelic-k8s-metrics-adapter/Chart.yaml
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Release workload charts
+        uses: helm/chart-releaser-action@v1.3.0
```

Comment:
Just out of curiosity. what would happen if the chart is released without updating the chart version?

Reply:
On CI it will break during the linting step. If you push to main directly it will refuse to update because the version exists.